powershell.html.markdown

```diff
@@ -13,116 +14,349 @@ Nearly all examples below can be a part of a shell script or executed directly
+$True  # => True
+$False  # => False
+
+# negate with !
+!$True   # => False
+!$False  # => True
+
+# Boolean Operators
+# Note "-and" and "-or" usage
+$True -and $False  # => False
+$False -or $True   # => True
+
+# True and False are actually 1 and 0 but only support limited arithmetic
+# However, casting the bool to int resolves this
+$True + $True # => 2
+$True * 8    # => '[System.Boolean] * [System.Int32]' is undefined
+[int]$True * 8 # => 8
+$False - 5   # => -5
+
+# Comparison operators look at the numerical value of True and False
+0 -eq $False  # => True
+1 -eq $True   # => True
+2 -eq $True   # => False
+-5 -ne $False # => True
+
+# Using boolean logical operators on ints casts them to booleans for evaluation
+# but their non-cast value is returned
+# Don't mix up with bool(ints) and bitwise -band/-bor
+[bool](0)     # => False
+[bool](4)     # => True
+[bool](-6)    # => True
+0 -band 2     # => 0
+-5 -bor 0     # => -5
+
+# Equality is -eq (equals)
+1 -eq 1  # => True
+2 -eq 1  # => False
+
+# Inequality is -ne (notequals)
+1 -ne 1  # => False
+2 -ne 1  # => True
+
+# More comparisons
+1 -lt 10  # => True
+1 -gt 10  # => False
+2 -le 2  # => True
+2 -ge 2  # => True
+
+# Seeing whether a value is in a range
+1 -lt 2 -and 2 -lt 3  # => True
+2 -lt 3 -and 3 -lt 2  # => False
+
+# (-is vs. -eq) -is checks if two objects are the same type
+# -eq checks if the objects have the same values.
+[System.Collections.ArrayList]$a = @()  # Point a at a new list
+$a = (1,2,3,4)
+$b = $a                                 # => Point b at what a is pointing to
+$b -is $a.getType()                     # => True, a and b equal same type
+$b -eq $a                               # => True, a and b values are equal
+[System.Collections.Hashtable]$b = @{}  # => Point a at a new hash table
+$b = @{'one' = 1 
+       'two' = 2}
+$b -is $a.GetType()                     # => False, a and b types not equal
+
+# Strings are created with " or ' but " is required for string interpolation
+"This is a string."
+'This is also a string.'
+
+# Strings can be added too! But try not to do this.
+"Hello " + "world!"  # => "Hello world!"
+
+# A string can be treated like a list of characters
+"Hello world!"[0]  # => 'H'
+
+# You can find the length of a string
+("This is a string").Length  # => 16
+
+# You can also format using f-strings or formatted string literals
+$name = "Steve"
+$age = 22
+"He said his name is $name." # => "He said his name is Steve"
+"{0} said he is {1} years old." -f $name, $age # => "Steve said he is 22 years old"
+"$name's name is $($name.Length) characters long." # => "Steve's name is 5 characters long."
+
+# $null is not an object
+$null  # => None
+
+# $null, 0, and empty strings and arrays all evaluate to False.
+# All other values are True
+function test ($value) {
+  if ($value) {Write-Output 'True'}
+    else {Write-Output 'False'}
+}
+test ($null) # => False
+test (0)   # => False
+test ("")  # => False
+test []  # => True
```

Comment:
I think this is confusing.

Reply:
I've added some comments and details to the objects.